src/Windows/Avalonia.Win32/WindowImpl.cs

```diff
@@ -643,15 +646,22 @@ public void Dispose()
                 _hwnd = IntPtr.Zero;
             }
 
-            if (_className != null)
+            if (isBeingDestroyed && _className != null)
```

Comment:
Is this check even necessary? We could just schedule `UnregisterWindowClass` from WM_DESTROY handler since it's always being called when window is destroyed.

Reply:
It made sense to me to semantically keep the clean up of the window class with the main part as `WndProc` isn't normally concerned with `_className` - or rather its window class - (even though both share the same class due to it being partial). As `Dispose()` gets called twice - first by `CloseInternal()` and then by `AppWndProc` on `WM_DESTROY`, we need to know which one did to schedule `UnregisterWindowClass` at the right time.

I'd be fine with going with your suggestion for the sake of simplicity though.